exchanges/bybit/bybit_test.go

```diff
@@ -2950,34 +2952,22 @@ func TestGetWithdrawalsHistory(t *testing.T) {
 
 func TestGetRecentTrades(t *testing.T) {
 	t.Parallel()
-	_, err := b.GetRecentTrades(context.Background(), spotTradablePair, asset.Spot)
-	if err != nil {
-		t.Error(err)
-	}
-	_, err = b.GetRecentTrades(context.Background(), inverseTradablePair, asset.CoinMarginedFutures)
-	if err != nil {
-		t.Error(err)
-	}
-	_, err = b.GetRecentTrades(context.Background(), usdtMarginedTradablePair, asset.USDTMarginedFutures)
-	if err != nil {
-		t.Error(err)
-	}
-	_, err = b.GetRecentTrades(context.Background(), usdcMarginedTradablePair, asset.USDCMarginedFutures)
-	if err != nil {
-		t.Error(err)
-	}
-	_, err = b.GetRecentTrades(context.Background(), spotTradablePair, asset.Futures)
-	if !errors.Is(err, asset.ErrNotSupported) {
-		t.Error(err)
-	}
-	cp, err := b.ExtractCurrencyPair("BTC-29DEC23-80000-C", asset.Options, true)
-	if err != nil {
-		t.Fatal(err)
-	}
-	_, err = b.GetRecentTrades(context.Background(), cp, asset.Options)
-	if err != nil {
-		t.Error(err)
+	for _, tt := range []struct {
+		a asset.Item
+		p currency.Pair
+	}{
+		{asset.Spot, spotTradablePair},
+		{asset.Options, optionsTradablePair},
+		{asset.CoinMarginedFutures, inverseTradablePair},
+		{asset.USDTMarginedFutures, usdtMarginedTradablePair},
+		{asset.USDCMarginedFutures, usdcMarginedTradablePair},
+	} {
+		_, err := b.GetRecentTrades(context.Background(), tt.p, tt.a)
+		assert.NoErrorf(t, err, "GetRecentTrades should not error for %s asset", tt.a)
```

Comment:
```suggestion
		t.Run(tt.a.String(), func(t *testing.T) {
			t.Parallel()
			_, err := b.GetRecentTrades(context.Background(), tt.p, tt.a)
			assert.NoErrorf(t, err, "GetRecentTrades should not error for %s asset", tt.a)
		})
```

Reply:
I'm not seeing the benefit of Run in this context; Only downsides:
* The test time is approximately the same or worse
* The verbose output is messy as hell

```
=== RUN   TestGetRecentTrades
=== PAUSE TestGetRecentTrades
=== CONT  TestGetRecentTrades
--- PASS: TestGetRecentTrades (0.04s)
PASS
ok      github.com/thrasher-corp/gocryptotrader/exchanges/bybit 0.215s
```

versus

```
=== RUN   TestGetRecentTrades
=== PAUSE TestGetRecentTrades
=== CONT  TestGetRecentTrades
=== RUN   TestGetRecentTrades/spot
=== PAUSE TestGetRecentTrades/spot
=== RUN   TestGetRecentTrades/options
=== PAUSE TestGetRecentTrades/options
=== RUN   TestGetRecentTrades/coinmarginedfutures
=== PAUSE TestGetRecentTrades/coinmarginedfutures
=== RUN   TestGetRecentTrades/usdtmarginedfutures
=== PAUSE TestGetRecentTrades/usdtmarginedfutures
=== RUN   TestGetRecentTrades/usdcmarginedfutures
=== PAUSE TestGetRecentTrades/usdcmarginedfutures
=== CONT  TestGetRecentTrades/spot
=== CONT  TestGetRecentTrades/usdtmarginedfutures
=== CONT  TestGetRecentTrades/coinmarginedfutures
=== CONT  TestGetRecentTrades/usdcmarginedfutures
=== CONT  TestGetRecentTrades/options
--- PASS: TestGetRecentTrades (0.00s)
    --- PASS: TestGetRecentTrades/spot (0.00s)
    --- PASS: TestGetRecentTrades/usdtmarginedfutures (0.01s)
    --- PASS: TestGetRecentTrades/coinmarginedfutures (0.02s)
    --- PASS: TestGetRecentTrades/options (0.03s)
    --- PASS: TestGetRecentTrades/usdcmarginedfutures (0.04s)
PASS
ok      github.com/thrasher-corp/gocryptotrader/exchanges/bybit 0.217s
```
```

Reply:
I found the opposite results in terms of timing. Running it over and over has some variability. I don't really find any issue with that output either. I also feel like the struct can have the `e` field to then remove the final scenario
```go
func TestGetRecentTrades(t *testing.T) {
	t.Parallel()
	for _, tt := range []struct {
		a             asset.Item
		p             currency.Pair
		e error
	}{
		{a: asset.Spot, p: spotTradablePair},
		{a: asset.Options, p: optionsTradablePair},
		{a: asset.CoinMarginedFutures, p: inverseTradablePair},
		{a: asset.USDTMarginedFutures, p: usdtMarginedTradablePair},
		{a: asset.USDCMarginedFutures, p: usdcMarginedTradablePair},
		{a: asset.Futures, p: spotTradablePair, e: asset.ErrNotSupported},
	} {
		t.Run(tt.a.String(), func(t *testing.T) {
			t.Parallel()
			_, err := b.GetRecentTrades(context.Background(), tt.p, tt.a)
			assert.ErrorIs(t, err, tt.expectedError, "GetRecentTrades unexpected error")
		})
	}
}
```
This is such a minor request and change though, so it can stay if you want 🤷

Reply:
Noted. I went your way, but it actually only saves 1 byte (even using `, nil` instead of named params),
and I feel the misleading `ErrorIs nil` and error message isn't worth that byte.

So yeah, I'll leave it as is. Thanks though, was worth exploring.